d3a_api_client/utils.py

```diff
@@ -264,3 +264,54 @@ def log_bid_offer_confirmation(message):
                          f"{round(energy, 2)} kWh at {price} cts/kWh")
     except Exception as e:
         logging.error(f"Logging bid/offer info failed.{e}")
+
+
+def flatten_info_dict(indict: dict) -> dict:
+    """
+    wrapper for _flatten_info_dict
+    """
+    if indict == {}:
+        return {}
+    outdict = {}
+    _flatten_info_dict(indict, outdict)
+    return outdict
+
+
+def _flatten_info_dict(indict: dict, outdict: dict):
+    """
+    Flattens market_info/tick_info information trees
+    outdict will hold references to all area subdicts of indict
+    """
+    for area_name, area_dict in indict.items():
+        outdict[area_name] = area_dict
+        if 'children' in area_dict:
+            _flatten_info_dict(indict[area_name]['children'], outdict)
+
+
+def get_uuid_from_area_name_in_tree_dict(area_name_uuid_mapping, name):
+    if name not in area_name_uuid_mapping:
+        raise ValueError(f"Could not find {name} in tree")
+    if len(area_name_uuid_mapping[name]) == 1:
+        return area_name_uuid_mapping[name][0]
+    else:
+        ValueError(f"There are multiple areas named {name} in the tree")
+
+
+def buffer_grid_tree_info(f):
+    @wraps(f)
+    def wrapper(self, message):
+        self.latest_grid_tree = message["grid_tree"]
+        self.latest_grid_tree_flat = flatten_info_dict(self.latest_grid_tree)
+        f(self, message)
+    return wrapper
+
+
+def create_area_name_uuid_mapping_from_tree_info(latest_grid_tree_flat: dict) -> dict:
+    area_name_uuid_mapping = {}
+    for area_uuid, area_dict in latest_grid_tree_flat.items():
+        if "area_name" in area_dict:
+            if area_uuid in area_name_uuid_mapping:
+                area_name_uuid_mapping[area_dict["area_name"]].append(area_uuid)
```

Comment:
As we agreed the are name should be unique in the grid , so why building the object as list of several areas as value ?

Reply:
We did not agree on that, AFAIR. 
You were the one that implemented the "unique names in the same parent area" , which is still the case and it is still allowed to have two areas named 'load' in the whole grid as long as they dont have the same parent.